src/vehicle.cpp

```diff
@@ -3896,6 +3898,26 @@ void vehicle::refresh()
                                 static_cast<int>( p ), svpv );
         relative_parts[pt].insert( vii, p );
 
+        if( vpi.has_flag( VPFLAG_OBSTACLE ) && vpi.has_flag( VPFLAG_OPAQUE ) ) {
```

Comment:
For readability, I think it makes sense to encapsulate this entire if statement in a function call.

Also, should be `if( vpi.has_flag( VPFLAG_OBSTACLE ) || vpi.has_flag( VPFLAG_OPAQUE ) )` - you want to catch all of windshields, fullboards, halfboards, and curtains or whatever else blocks like of sight without blocking movement.